src/envoy/http/authn/authn_utils.cc

```diff
@@ -56,9 +58,10 @@ bool JsonIterCallback::JsonIteratorCallback(const std::string& key,
         payload->add_audiences(aud_vector[i]);
       }
     } else {
-      // will throw execption if value type is not string.
+      // Will throw execption if value type is not string.
       // In current implementation, only string objects are extracted into
-      // claims.
+      // claims. If call obj.asJsonString(), will get "panic: not reached" from
+      // json_loader.cc.
       (*claims)[key] = obj.asString();
```

Comment:
It would be cleaner to do this in a separate try block on top. Then the second part is just to extract aud. It's avoid nested try catch as well as easy to see all string claims are kept.